sdcm/utils/net.py

```diff
@@ -21,16 +21,18 @@ def get_my_ip():
 
 
 def get_my_public_ip() -> str:
-    hostnames = ['https://api4.my-ip.io/ip', "https://api.ipify.org", 'https://ip4.seeip.org/ip',
+    hostnames = ['https://api4.my-ip.io/ip', 'https://api.ipify.org', 'https://ip4.seeip.org/ip',
                  'http://ipv4.icanhazip.com']
 
-    for hostname in hostnames:
-        result = requests.get(hostname, timeout=10)
-        if result.ok:
-            try:
-                ip_address = result.text.strip()
-                ipaddress.ip_address(ip_address)  # validating that we got IP address
-                return ip_address
-            except ValueError:
-                continue
-    return ""
+    ip_address = "[Not Found]"
+    for idx, hostname in enumerate(hostnames):
+        try:
+            result = requests.get(hostname, timeout=10)
+            result.raise_for_status()
+
+            ip_address = result.text.strip()
+            ipaddress.ip_address(ip_address)  # validating that we got IP address
+        except (ValueError, requests.exceptions.RequestException):
```

Comment:
If it will be other exception, we also want to continue to next host, no?

Reply:
it's 3 lines of code, it would be very hard for it to generate any other errors.

Reply:
Maybe not. I just think that it would be right to "close the circle"